pipelines/common/src/main/java/com/google/fhir/analytics/DwhFiles.java

```diff
@@ -212,21 +215,27 @@ public Set<String> findNonEmptyFhirResourceTypes() throws IOException {
           fileSet.add(metadata.resourceId().getCurrentDirectory().getFilename());
         }
       } else if (matchResult.status() == Status.ERROR) {
-        log.error("Error matching resource types under {} ", dwhRoot);
-        throw new IOException(String.format("Error matching resource types under %s", dwhRoot));
+        log.error("Error matching {} under {} ", fileType, dwhRoot);
+        throw new IOException(String.format("Error matching %s under %s", fileType, dwhRoot));
       }
     }
 
     Set<String> typeSet = Sets.newHashSet();
     for (String file : fileSet) {
-      try {
-        fhirContext.getResourceType(file);
-        typeSet.add(file);
-      } catch (DataFormatException e) {
-        log.debug("Ignoring file {} which is not a FHIR resource.", file);
+      if (viewFlag) {
+        try {
+          fhirContext.getResourceType(file);
+          typeSet.add(file);
+        } catch (DataFormatException e) {
+          log.debug("Ignoring file {} which is not a FHIR resource.", file);
+        }
+      } else {
+        if (file.toLowerCase().equals(file)) {
```

Comment:
I think this needs some more validation (similar to resource types), e.g., checking that the file actually matches the expected dir name for a view. There can be other directories under DWH root, e.g., `incremental_run` or dirs corresponding to failed incremental runs.

Reply:
Added logic to check if the file directory contains parquet files before adding to the result `Set`

Reply:
Commented above; I don't think this logic for checking that the dir has parquet files adds much; we already know the dir is non-empty because of `math` on line 207 above.

Reply:
Done. See the above comment on this issue.